### PR TITLE
Allow piped stdin in run_subprocess_with_logging

### PR DESCRIPTION
**Pull request by dliappis**
Sometimes we need to used as stdin the output from another command
(e.g. pass secure settings to elasticsearch-keystore tool is
one example).

This commit optionally allows passing stdin from another subprocess
to run_subprocess_with_logging.

### CODE REVIEW

File: esrally/utils/process.py [hunk 1]
```diff
@@ -60,13 +60,15 @@ def exit_status_as_bool(runnable, quiet=False):
         return False
 
 
-def run_subprocess_with_logging(command_line, header=None, level=logging.INFO, env=None):
+def run_subprocess_with_logging(command_line, header=None, level=logging.INFO, stdin=None, env=None):
```
Comment (dliappis) [inline esrally/utils/process.py]: I didn't find any existing tests for this, so deferred -- for now -- writing new ones.

**Comment by dliappis** [comment]
Rebased https://github.com/elastic/rally/pull/859 into this one.

File: esrally/utils/process.py [hunk 2]
```diff
@@ -60,29 +60,46 @@ def exit_status_as_bool(runnable, quiet=False):
...
-                has_output = False
+
+    # pylint: disable=subprocess-popen-preexec-fn
+    with subprocess.Popen(command_line_args,
+                          stdout=subprocess.PIPE,
+                          stderr=subprocess.STDOUT,
+                          env=env,
+                          stdin=stdin if stdin else None,
+                          preexec_fn=pre_exec) as command_line_process:
+        if stdin:
```
[the remark this answers is not in the document]
Comment (dliappis) [reply esrally/utils/process.py]: [#communicate()](https://docs.python.org/3.8/library/subprocess.html#subprocess.Popen.communicate) will wait until the process completes. As a result the existing code (including looping through `command_line_process.stdout.readline()` and "streaming" the results) won't work. 

Or are you suggesting replacing the everything we currently have within the `with:` block with the above 3 lines?

[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply esrally/utils/process.py]: I am suggesting that we replace the whole block with the above three lines. Given the we use it only for short-lived subprocesses anyway, I think it is ok to "lose" the streaming aspect.

[the remark this answers is not in the document]
Comment (dliappis) [reply esrally/utils/process.py]: Ok! With this approach in mind I pushed https://github.com/elastic/rally/pull/863/commits/06316615b4ace54892b4b83d90e5c78830892f5f

**Pull request merged**